src/main/java/com/bootcamp/demo/model/Card.java

```diff
@@ -0,0 +1,113 @@
+package com.bootcamp.demo.model;
+
+import java.util.Calendar;
+
+public class Card extends Throwable{
```

Comment:
A model class should be a simple POJO containing only fields, constructor, setters and getters (and other inherited methods like toString). Please extract all the validation logic in a separate bean, maybe a validation service that could be called by the RestController when handling/receiving a Card. This approach adheres to the single responsibility principle and allows the validators to evolve separately (e.g. be replaced by a dedicated validation library).